### PR TITLE
devops: add chromium-headless-shell bots

### PR DESCRIPTION
**Pull request by mxschmitt**
https://github.com/microsoft/playwright/issues/33144

### CODE REVIEW

File: .github/workflows/tests_primary.yml [hunk 1]
```diff
@@ -223,3 +223,25 @@ jobs:
         flakiness-client-id: ${{ secrets.AZURE_FLAKINESS_DASHBOARD_CLIENT_ID }}
         flakiness-tenant-id: ${{ secrets.AZURE_FLAKINESS_DASHBOARD_TENANT_ID }}
         flakiness-subscription-id: ${{ secrets.AZURE_FLAKINESS_DASHBOARD_SUBSCRIPTION_ID }}
+
+  test_linux_chromium_headless_shell:
```
Comment (mxschmitt) [inline .github/workflows/tests_primary.yml:227]: Note: I'll move it over to secondary before I merge.

File: .github/workflows/tests_primary.yml [hunk 2]
```diff
@@ -223,3 +223,25 @@ jobs:
...
+    - uses: ./.github/actions/run-test
+      with:
+        browsers-to-install: chromium-headless-shell
+        command: npm run ctest
+        bot-name: "headless-shell-${{ matrix.runs-on }}"
+        flakiness-client-id: ${{ secrets.AZURE_FLAKINESS_DASHBOARD_CLIENT_ID }}
+        flakiness-tenant-id: ${{ secrets.AZURE_FLAKINESS_DASHBOARD_TENANT_ID }}
+        flakiness-subscription-id: ${{ secrets.AZURE_FLAKINESS_DASHBOARD_SUBSCRIPTION_ID }}
+      env:
+        PLAYWRIGHT_CHROMIUM_USE_HEADLESS_NEW: 1
```
Comment (dgozman) [inline .github/workflows/tests_primary.yml:246]: Drop this line.

File: tests/library/browsercontext-credentials.spec.ts [hunk 3]
```diff
@@ -18,8 +18,8 @@
 import { browserTest as base, expect } from '../config/browserTest';
 
 const it = base.extend<{ isChromiumHeadedLike: boolean }>({
-  isChromiumHeadedLike: async ({ browserName, headless }, use) => {
-    const isChromiumHeadedLike = browserName === 'chromium' && (!headless || !!process.env.PLAYWRIGHT_CHROMIUM_USE_HEADLESS_NEW);
+  isChromiumHeadedLike: async ({ browserName, headless, channel }, use) => {
```
Comment (dgozman) [inline tests/library/browsercontext-credentials.spec.ts:21]: I don't think this change is needed. We should never run chromium-headless-shell in headed mode.

File: tests/library/headful.spec.ts [hunk 4]
```diff
@@ -191,8 +191,9 @@ it('should not block third party SameSite=None cookies', async ({ httpsServer, b
   await page.close();
 });
 
-it('should not override viewport size when passed null', async function({ browserName, server, browser }) {
+it('should not override viewport size when passed null', async function({ browserName, server, browser, channel }) {
   it.skip(browserName === 'webkit', 'Our WebKit embedder does not respect window features');
+  it.fixme(channel === 'chromium-headless-shell', 'outerWidth does not get updated after resizeTo');
```
Comment (dgozman) [inline tests/library/headful.spec.ts:196]: Skip this whole file instead.

**Review by dgozman — APPROVED** [review]

**Pull request merged**